src/renderer/helpers/api/local.js

```diff
@@ -166,6 +168,115 @@ function decipherFormats(formats, player) {
   }
 }
 
+export async function getLocalChannelId(url) {
+  try {
+    const innertube = await createInnertube()
+
+    // resolveURL throws an error if the URL doesn't exist
+    const navigationEndpoint = await innertube.resolveURL(url)
+
+    if (navigationEndpoint.metadata.page_type === 'WEB_PAGE_TYPE_CHANNEL') {
+      return navigationEndpoint.payload.browseId
+    } else {
+      return null
+    }
+  } catch {
+    return null
+  }
+}
+
+/**
+ * Returns the channel or the channel termination reason
+ * @param {string} id
+ */
+export async function getLocalChannel(id) {
+  const innertube = await createInnertube()
+  let result
+  try {
+    result = await innertube.getChannel(id)
+  } catch (error) {
+    if (error instanceof ChannelError) {
+      result = error.message
+    } else {
+      throw error
+    }
+  }
+  return result
+}
```

Comment:
For return result, maybe returning object on error case and on the caller to check if it's error is much more readable than checking the return value is string (i.e. error message returned)?

Reply:
Agreed `return { error: error.message }` would make it more readable instead of checking if it's a string